worker/reboot/reboot.go

```diff
@@ -29,18 +27,12 @@ const RebootMessage = "preparing for reboot"
 type Reboot struct {
 	tomb        tomb.Tomb
 	st          reboot.State
-	tag         names.MachineTag
 	machineLock *fslock.Lock
 }
 
-func NewReboot(st reboot.State, agentConfig agent.Config, machineLock *fslock.Lock) (worker.Worker, error) {
-	tag, ok := agentConfig.Tag().(names.MachineTag)
-	if !ok {
-		return nil, errors.Errorf("Expected names.MachineTag, got %T: %v", agentConfig.Tag(), agentConfig.Tag())
-	}
+func NewReboot(st reboot.State, machineLock *fslock.Lock) (worker.Worker, error) {
```

Comment:
pretty sure it's not in this branch yet, otherwise it wouldn't have been there to begin with.